packages/@aws-cdk/toolkit-lib/lib/api/drift/drift-formatter.ts

```diff
@@ -0,0 +1,146 @@
+import { format } from 'node:util';
+import * as cxschema from '@aws-cdk/cloud-assembly-schema';
+import { formatStackDriftChanges } from '@aws-cdk/cloudformation-diff';
+import type * as cxapi from '@aws-cdk/cx-api';
+import type { DescribeStackResourceDriftsCommandOutput } from '@aws-sdk/client-cloudformation';
+import * as chalk from 'chalk';
+import { StringWriteStream } from '../streams';
+import type { IDriftIoHelper } from './interfaces';
+
+/**
+ * Output of formatStackDrift
+ */
+export interface FormatStackDriftOutput {
+  /**
+   * Number of resources with drift
+   */
+  readonly numResourcesWithDrift?: number;
+
+  /**
+   * How many resources were not checked for drift
+   */
+  readonly numResourcesUnchecked?: number;
+
+  /**
+   * Complete formatted drift
+   */
+  readonly formattedDrift: string;
+}
+
+/**
+ * Props for the Drift Formatter
+ */
+export interface DriftFormatterProps {
+  /**
+   * Helper for IO operations
+   */
+  readonly ioHelper: IDriftIoHelper;
+
+  /**
+   * The CloudFormation stack artifact
+   */
+  readonly stack: cxapi.CloudFormationStackArtifact;
+
+  /**
+   * The results of stack drift detection
+   */
+  readonly driftResults?: DescribeStackResourceDriftsCommandOutput;
+
+  /**
+   * List of all stack resources. Used to determine what resources weren't checked
+   * for drift.
+   */
+  readonly allStackResources?: Map<string, string>;
+}
+
+/**
+ * Properties specific to formatting the stack drift
+ */
+export interface FormatStackDriftOptions {
+  /**
+   * Silences 'No drift detected' messages
+   *
+   * @default false
+   */
+  readonly quiet?: boolean;
+
+  /**
+   * Whether to be verbose
+   *
+   * @default false
+   */
+  readonly verbose?: boolean;
+}
```

Comment:
I'm aware this is just copying what `diff` currently does for legacy reasons. But I don't want to carry this forward to a brand new API. If you are really keen on having this, let's get on a call together to figure out a solution. Otherwise just drop this please.

Reply:
Yup, was mainly just copying over the flags and functionality from`diff`. I'm not particularly attached to them if that's not the direction we want for the new API, so I'm happy to drop them. Though in the case of `verbose`, I do want to keep that functionality of showing all resources, including those unchecked (addressing [this concern](https://github.com/aws/aws-cdk-cli/pull/442#discussion_r2081366824)) but maybe to prevent overloading the flag, we can rename `--verbose` to `--show-unchecked` in this command, or something along those lines?